pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java

```diff
@@ -0,0 +1,135 @@
+ * under the License.
+ */
+package org.apache.pinot.plugin.minion.tasks.purge;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.pinot.common.data.Segment;
+import org.apache.pinot.common.metadata.segment.SegmentZKMetadata;
+import org.apache.pinot.controller.helix.core.minion.ClusterInfoAccessor;
+import org.apache.pinot.controller.helix.core.minion.generator.BaseTaskGenerator;
+import org.apache.pinot.controller.helix.core.minion.generator.TaskGeneratorUtils;
+import org.apache.pinot.core.common.MinionConstants;
+import org.apache.pinot.core.minion.PinotTaskConfig;
+import org.apache.pinot.spi.annotations.minion.TaskGenerator;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableTaskConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+@TaskGenerator
+public class PurgeTaskGenerator extends BaseTaskGenerator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PurgeTaskGenerator.class);
+
+    private static final PurgeTaskGenerator INSTANCE = new PurgeTaskGenerator();
+
+    private ClusterInfoAccessor _clusterInfoAccessor;
+
+    public static Object newInstance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public void init(ClusterInfoAccessor clusterInfoAccessor) {
+        LOGGER.info("Initializing purge task");
+        _clusterInfoAccessor = clusterInfoAccessor;
+    }
+
+
+    @Override
+    public String getTaskType() {
+        return MinionConstants.PurgeTask.TASK_TYPE;
+    }
+
+
+
+    @Override
+    public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
+        String taskType = MinionConstants.PurgeTask.TASK_TYPE;
+        List<PinotTaskConfig> pinotTaskConfigs = new ArrayList<>();
+
+        for (TableConfig tableConfig : tableConfigs) {
+
+            String tableName = tableConfig.getTableName();
+            if (tableConfig.getTableType() != TableType.REALTIME) {
+                LOGGER.warn("Skip generating task: {} for non-REALTIME table: {}", taskType, tableName);
+                continue;
+            }
+            TableTaskConfig tableTaskConfig = tableConfig.getTaskConfig();
+            Preconditions.checkNotNull(tableTaskConfig);
+            Map<String, String> taskConfigs =
+                    tableTaskConfig.getConfigsForTaskType(MinionConstants.PurgeTask.TASK_TYPE);
+            try {
+                Preconditions.checkNotNull(taskConfigs, "Task config shouldn't be null for Table: {}", tableName);
+            } catch (Exception e) {
+                continue;
+            }
+            LOGGER.info("Start generating task configs for table: {} for task: {}", tableName, taskType);
+
+            if (tableConfig.getTableType() == TableType.REALTIME) {
```

Comment:
And this logic seems contradicting with the logic above as it skips for realtime tables.